repo/grpc_repository_client.go

```diff
@@ -321,23 +329,34 @@ func (r *grpcInnerSession) PutManifest(ctx context.Context, labels map[string]st
 	return "", errNoSessionResponse()
 }
 
+func (r *grpcRepositoryClient) SetFindManifestPageSizeForTesting(v int32) {
+	r.findManifestsPageSize = v
+}
+
 func (r *grpcRepositoryClient) FindManifests(ctx context.Context, labels map[string]string) ([]*manifest.EntryMetadata, error) {
 	return maybeRetry(ctx, r, func(ctx context.Context, sess *grpcInnerSession) ([]*manifest.EntryMetadata, error) {
-		return sess.FindManifests(ctx, labels)
+		return sess.FindManifests(ctx, labels, r.findManifestsPageSize)
 	})
 }
 
-func (r *grpcInnerSession) FindManifests(ctx context.Context, labels map[string]string) ([]*manifest.EntryMetadata, error) {
+func (r *grpcInnerSession) FindManifests(ctx context.Context, labels map[string]string, pageSize int32) ([]*manifest.EntryMetadata, error) {
+	var entries []*manifest.EntryMetadata
+
 	for resp := range r.sendRequest(ctx, &apipb.SessionRequest{
 		Request: &apipb.SessionRequest_FindManifests{
 			FindManifests: &apipb.FindManifestsRequest{
-				Labels: labels,
+				Labels:   labels,
+				PageSize: pageSize,
 			},
 		},
 	}) {
 		switch rr := resp.Response.(type) {
 		case *apipb.SessionResponse_FindManifests:
-			return decodeManifestEntryMetadataList(rr.FindManifests.GetMetadata()), nil
+			entries = appendManifestEntryMetadataList(entries, rr.FindManifests.GetMetadata())
+
+			if !resp.GetHasMore() {
+				return entries, nil
+			}
 
 		default:
 			return nil, unhandledSessionResponse(resp)
```

Comment:
nit; might be helpful for debugging later if we include the count of the entries read so far in the error message